src/storage/trie/trie_node.rs

```diff
@@ -122,15 +124,25 @@ impl TrieNode {
         for char in db_trie_node.child_chars {
             children.insert(
                 char as u8,
-                TrieNodeType::Serialized(SerializedTrieNode::new(None)),
+                TrieNodeType::Serialized(SerializedTrieNode::new()),
```

Comment:
If the struct is empty, do we actually need the struct?

Reply:
left a TODO comment